README.md

```diff
@@ -517,6 +517,266 @@ if err != nil {
 // From now on, you can use response to continue with a new flow. You will notice here that you have a new `stateHandle` which signals a new flow. Your `interaction_handle` will remain the same.
 ```
 
+#### Self Service Registration
+
+```go
+	var response *Response
+
+    client, err := NewClient(
+        WithClientID("{CLIENT_ID}"),
+        WithClientSecret("{CLIENT_SECRET}"),        // Required for confidential clients.
+        WithIssuer("{ISSUER}"),                     // e.g. https://foo.okta.com/oauth2/default, https://foo.okta.com/oauth2/busar5vgt5TSDsfcJ0h7
+        WithScopes([]string{"openid", "profile"}),  // Must include at least `openid`. Include `profile` if you want to do token exchange
+        WithRedirectURI("{REDIRECT_URI}"),          // Must match the redirect uri in client app settings/console
+    )
+	if err != nil {
+		panic(err)
+	}
+
+	interactHandle, err := client.Interact(context.TODO())
```

Comment:
I guess you'll update all code samples at once when you refactor interact to return a context.

Reply:
Yes.  That is the plan, since this is the way it works right now